dll/ntdll/ldr/ldrinit.c

```diff
@@ -1636,8 +1636,13 @@ LdrpInitializeProcessCompat(PVOID pProcessActctx, PVOID* pOldShimData)
     if (ContextCompatInfo->ElementCount == 0)
         return;
 
-    /* Search for known GUID's, starting from newest to oldest. */
-    for (cur = 0; cur < RTL_NUMBER_OF(KnownCompatGuids); ++cur)
+    /* Search for known GUID's, starting from oldest to newest.
+       Note that on Windows it is somewhat reversed, starting from the latest known
+       vesion, going down. But we are not Windows, trying to allow a lower version,
+       we are ReactOS trying to fake a higher version. So we interpret what Windows
+       does as "try the closes version to the actual version", so we start with the
+       lowest version, which is closest to Windows 2003, which we mostly are. */
+    for (cur = RTL_NUMBER_OF(KnownCompatGuids) - 1; cur != -1; --cur)
```

Comment:
This is fine, however the most popular pattern for a reverse loop in C seems to be:
```suggestion
    for (cur = RTL_NUMBER_OF(KnownCompatGuids) - 1; --cur >= 0;)
```

Change at will.

Reply:
This wouldn't work, because cur is unsigned. It's also broken, because it starts with ```cur = RTL_NUMBER_OF(KnownCompatGuids) - 2```. Additionally it breaks the logic in combination with continue and break. And I find it ugly :)